client/packages/system/src/Location/ListView/LocationEditModal/LocationEditModal.tsx

```diff
@@ -139,6 +141,15 @@ export const LocationEditModal: FC<LocationEditModalProps> = ({
             label={t('label.code')}
             InputLabelProps={{ shrink: true }}
           />
+          <ColdStorageTypeInput
+            value={draft.coldStorageType ?? null}
+            label={t('label.temperature')}
+            onChange={coldStorageType =>
+              onUpdate({
+                coldStorageType: coldStorageType as ColdStorageTypeNode,
```

Comment:
I always seem to have trouble with this. The incoming type is `ColdStorageTypeFragment`, which is basically the same as `ColdStorageTypeNode` except is has a subset of fields. It's fine in this case, as only the `id` actually gets passed through to the mutation, but would like to do it without needing `as` if possible?

Reply:
I think the trick is to use the same fragment in both graphql queries, so that types match.
Here's my suggestion branch: https://github.com/msupply-foundation/open-msupply/compare/4367(3)-edit-storage-on-location-modal...4367-suggestion-remove-as

However, I don't like importing the fragment from a different graphql file, seems to make the `../../../../` issue when doing `yarn generate` even worse.
Would be better to move the `location_types` query from `item` to `location` I think?

I also had to bring through the min and max temperature and do other updates to avoid other issues where those were it being used.

So possibly if you just did added min_max to the `ColdStorageTypeFragment` it might compile? I think the types should then be equivalent? Haven't tried though.

The idea of using a more narrow fragment feels good in that you only bring through the data you actually need, but does cause other headaches, so probably only worth trying to be frugal if we've got a lot of loaders or something that genuinely slow things down...

I'm sure there's other ways we could deal with this too.

Reply:
> So possibly if you just did added min_max to the `ColdStorageTypeFragment` it might compile? I think the types should then be equivalent? Haven't tried though.

Yeah that did the trick, thanks :) 

6f69e19b68